### ISSUE TITLE
avds should be changed to avd

### ISSUE THREAD

**Issue by Shrif88** [selendroid/selendroid#1134]
https://github.com/selendroid/selendroid/blob/f93882149ae37dca611d7cbc732aac7b18e597b9/selendroid-standalone/src/main/java/io/selendroid/standalone/android/impl/DefaultAndroidEmulator.java#L148

**Comment by alexcomsa** [comment]
This may depended on the SDK version you are using, also maybe take in to account that starting with Android SDK version 25.2.3  the android command is deprecated and you need to use [avdmanager](https://developer.android.com/studio/command-line/avdmanager.html) to manage all virtual devices.
Maybe something like this : 
```java
 if (AndroidSdk.getAndroidVersionNumber() > SDK_VERSION_FOR_AVD_MANAGER) {

            getAVDCommand = new CommandLine(AndroidSdk.avdManager());
            getAVDCommand.addArgument("list", false);

        } else {
            getAVDCommand = new CommandLine(AndroidSdk.android());
            getAVDCommand.addArgument("list", false);
            getAVDCommand.addArgument("avds", false);
        }
```

**Comment by rgoerner** [comment]
Patched version (no guarantee) can be found here: https://ufile.io/ku8bl

**Comment by rgoerner** [comment]
@alexcomsa
Do you have an idea? I exchanged the code like your example to:

  public static List<AndroidEmulator> listAvailableAvds() throws AndroidDeviceException {
    List<AndroidEmulator> avds = Lists.newArrayList();

    CommandLine cmd = new CommandLine(AndroidSdk.avdManager());
    cmd.addArgument("list", false);

    String output = null;


I receive a compile error 

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.3.2:compile (default-compile) on project selendroid-standalone: Compilation failure
[ERROR] /usr/src/selendroid-0.17.0/selendroid-standalone/src/main/java/io/selendroid/standalone/android/impl/DefaultAndroidEmulator.java:[146,48] error: cannot find symbol

line 146 is the line with CommandLine cmd = new CommandLine(AndroidSdk.avdManager());

Maybe you have an idea? I am not a Java guy ;( Thanks

**Comment by alexcomsa** [comment]
Yes because AndroidSdk.avdManager() does not exist in the master branch.

**Comment by alexcomsa** [comment]
I can push the branch later if you need it. You can try to build the standalone from it.

**Comment by rgoerner** [comment]
How can I modify / change the code? Can you maybe post the code here or so? I am really no Java guy ;( at all :( I just try to get this working. Many thanks for your help.

**Comment by rgoerner** [comment]
Maybe you have a working bin for me? This would be the best solution, maybe.
I am sad that this project does not seem to be maintained anymore or ?

**Comment by alexcomsa** [comment]
@rgoerner 
I'm not sure about the status of the project maybe @DominikDary can shed some light about it.

You can try to use this build  https://nofile.io/f/0gDOm6rnX3h/selendroid-04.jar
to see if it works for you .
If you use an android sdk version >= 25.3.0 you will need to pass as a command line argument the full path to the avdmanager command line tool
`java  -jar selendroid-04.jar  -app your_test_application.apk -avdManagerHome /usr/local/bin/avdmanager`

**Comment by rgoerner** [comment]
@alexcomsa 
I Think in general, to use this, something has to be changed in selendroid-standlone/src/main/java/io/selendroid/standalone/android/AndroidSdk.java or?

There is a missing entry for avdManager there. How did you modify this?

**Comment by rgoerner** [comment]
I added:

public static File avdManager() {
    StringBuffer command = new StringBuffer();
    command.append(toolsHome());
    return new File(toolsHome(), "avdManager" + platformExecutableSuffixBat());
  }

**Comment by rgoerner** [comment]
I think avdManager list ist also not the correct setting, because, this does not list just the running devices, but all available images ;-<

**Comment by rgoerner** [comment]
The correct command is: adb devices

This will show you a list with the devices, the real one connected, and the emulators, which are setup, not the ones, which are available - this is not correct. So adb devices is correct setting. avdManager list or something like that will just list *all* the available devices, but you need a list of the available and running ones :-)

**Comment by rgoerner** [comment]
location is
sdk\platform-tools\adb.exe

**Comment by rgoerner** [comment]
I checked the code again, a command for the adb devices is already in place.
Error seems to be more here in this class:

selendroid-standalone/src/main/java/io/selendroid/standalone/server/model/SelendroidStandaloneDriver.java

code is where the error occurs here:

**Comment by rgoerner** [comment]
I am sure, this is a problem with the class public synchronized JSONArray getSupportedDevices() - but I am not a programer either to debug this ;-( does anyone got it solved or has the same issue maybe?

**Comment by marcelogomez** [comment]
@alexcomsa is addressing this in #1136  :)

**Comment by lpetre** [comment]
Fixed in #1136
